### PR TITLE
Add FP16 option in save_combine_op

### PR DESCRIPTION
**Pull request by sidgoyal78**
Issue: it fails the check since the 3rd tensor has size > 2050.

### CODE REVIEW

File: paddle/fluid/operators/save_load_combine_op_test.cc [hunk 1]
```diff
@@ -123,15 +127,92 @@ TEST(SaveLoadCombineOp, CPU) {
...
+
+// FP16 version of SaveLoadCombineOp Test
+TEST(SaveLoadCombineFP16Op, CPU) {
+  paddle::framework::Scope scope;
+  paddle::platform::CPUPlace place;
+
+  std::vector<int> lod1 = {0, 1, 2, 3};
+  int numel1 = 100;
+  paddle::framework::LoD expect_lod1;
+  float* expect1 = CreateForSaveCombineOp<float>(10, 10, lod1, "test_var1",
```
Comment (kexinzhao) [inline paddle/fluid/operators/save_load_combine_op_test.cc]: Not sure if it is related to the error. 
But the last number of your lod vector info should match the first dimension of your tensor, which is 10 here.

File: paddle/fluid/operators/save_load_combine_op_test.cc [hunk 2]
```diff
@@ -123,15 +127,92 @@ TEST(SaveLoadCombineOp, CPU) {
...
+  std::vector<int> lod2 = {0, 2, 5, 10};
+  int numel2 = 200;
+  paddle::framework::LoD expect_lod2;
+  float* expect2 = CreateForSaveCombineOp<float>(10, 20, lod2, "test_var2",
+                                                 place, &scope, &expect_lod2);
+
+  std::vector<int> lod3 = {0, 1, 2, 3};
+  int numel3 = 2050;
+  paddle::framework::LoD expect_lod3;
+  float* expect3 = CreateForSaveCombineOp<float>(1, 2050, lod3, "test_var3",
```
Comment (kexinzhao) [inline paddle/fluid/operators/save_load_combine_op_test.cc]: Your lod3 = {0, 1, 2, 3}, but the first dimension of your tensor is 1. Not sure if this is right.

File: paddle/fluid/operators/save_load_combine_op_test.cc (hunk 1 above)
Comment (sidgoyal78) [reply paddle/fluid/operators/save_load_combine_op_test.cc]: Fixed (though it is not related to the error).

File: paddle/fluid/operators/save_load_combine_op_test.cc (hunk 2 above)
Comment (sidgoyal78) [reply paddle/fluid/operators/save_load_combine_op_test.cc]: Fixed.

**Pull request merged**